echopype/calibrate/ek80_complex.py

```diff
@@ -258,6 +259,37 @@ def get_transmit_signal(
     return y_all, y_time_all
 
 
+def _convolve_per_channel(backscatter_subset: np.ndarray, replica_dict: dict, channels: dict):
+    """
+    Convolve `backscatter_subset` array along range sample dimension for each channel.
+    The `backscatter_subset` array is a numpy array and has implicit dimensions
+    `('range_sample', 'channel')`.
+
+    When the `backscatter_subset` array is all 0s, we return it since the resulting
+    convolution will be all 0s, irrespective of what the corresponding transmit
+    signal is.
+
+    When this function is used in `compress_pulse`, the array that is being sent
+    as backscatter subset corresponds to a specific `ping_time` and `beam`, from
+    the backscatter array.
+    """
+    # Return if all 0s
+    if np.all(backscatter_subset == 0.0 + 0.0j):
+        return backscatter_subset
+    else:
+        # Create zeros like array from `backscatter_subset`
+        convolved = np.zeros_like(backscatter_subset, dtype=np.complex64)
+        # Iterate over channels
+        for i, channel in enumerate(channels):
+            # Extract replica values
+            replica = replica_dict[str(channel.values)]
+            # Convolve backscatter and chirp replica
+            convolved[:, i] = signal.convolve(backscatter_subset[:, i], replica, mode="full")[
+                replica.size - 1 :
+            ]
```

Comment:
This is tiny thing but I wonder it may be good to change `i` to maybe `ch_seq` or something, so that it is clear what that is, and also does not run into the confusion with the imaginary part being i or j, etc.

Reply:
I agree I'll make this change